### PR TITLE
fix(input-date-picker): reset active date picker date after closing

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #6495 

## Summary

This ensures the active date is reset properly after a `calcite-input-date-picker`'s date picker is closed (either by blurring or selecting a date).

## Notes

* This adds an internal `reset` method that can be made public in the future if needed. 
* There was an existing `reset` method prior to these changes that seems to clear the active date when the date picker is blurred or when `Escape` is pressed. We should revisit this behavior since it was [added when the input and date picker were a single component](https://github.com/Esri/calcite-components/blob/v1.0.0-beta.22/src/components/calcite-date/calcite-date.tsx#L263-L270) and it does seem odd to have the date-picker reset in these scenarios. cc @macandcheese @SkyeSeitz @ashetland 
* Utility test methods to interact with internal components were added and tests were updated to leverage them
* Slightly increases the duration factor in the tests to work around #6604

### CODE REVIEW

File: packages/calcite-components/src/components/input-date-picker/input-date-picker.e2e.ts [hunk 1]
```diff
@@ -294,38 +345,30 @@ describe("calcite-input-date-picker", () => {
...
-              .shadowRoot.querySelector("calcite-input")
-              .shadowRoot.querySelector("input").value
-        );
 
       await page.keyboard.press("Tab");
       await page.keyboard.type("1/");
       await page.waitForChanges();
 
-      expect(await getInputValue()).toBe("١‏/");
+      expect(await getDateInputValue(page)).toBe("١‏/");
```
Comment (benelan) [inline packages/calcite-components/src/components/input-date-picker/input-date-picker.e2e.ts:353]: thanks for hanging that up to DRY!

File: packages/calcite-components/src/tests/utils.ts [hunk 2]
```diff
@@ -268,7 +268,7 @@ export async function newProgrammaticE2EPage(): Promise<E2EPage> {
  */
 export async function skipAnimations(page: E2EPage): Promise<void> {
   await page.addStyleTag({
-    content: `:root { --calcite-duration-factor: 0; }`
+    content: `:root { --calcite-duration-factor: 0.01; }`
```
Comment (benelan) [inline packages/calcite-components/src/tests/utils.ts:271]: Can you explain why changing this to 0.1 is required, is it a chromatic thing?

**Review by benelan — APPROVED** [review]
LGTM

File: packages/calcite-components/src/tests/utils.ts (hunk 2 above)
Comment (jcfranco) [reply packages/calcite-components/src/tests/utils.ts:271]: Sure. It's to work around [this issue](https://github.com/Esri/calcite-components/issues/6604).

Good catch though. I'll add a comment to this for context.

**Comment by jcfranco** [comment]
Extracted the restoring of missing focus assertions to a separate PR: https://github.com/Esri/calcite-components/pull/7235

**Pull request merged**